doc/rm/c_cpp_coding_style.md

```diff
@@ -110,6 +110,20 @@ This convention helps readers distinguish which files they should not expect to
 
 The above rules also do not apply to system includes, which should be included by the names dictated by the ISO standard, e.g. `#include <stddef.h>`.
 
+### Linker Script Provided Definitions
+
+Some C/C++ programs may need to use symbols that are defined by a linker script.
+Referring to linker script symbols can be complex and error-prone, as they don't quite work like C's global variables.
+
+If you need to refer to the linker-script-defined symbol `_my_linker_symbol`, the correct way to do so is with an incomplete extern char array declaration, as shown below.
+This allows you to treat the value as a pointer, without C expecting storage to be allocated for the symbol.
```

Comment:
There's one more key thing: don't dereference it, as you'll get garbage. (e.g. for cases where you have `_something = SIZEOF(.section)`)

Reply:
I think it's worth underscoring that the "value" of the symbol is its *address*, which is easy to mix up if you've never seen this pattern before.